### DISCUSSION TITLE
Input does not register spacebar key presses when inside RadioGroup

### DISCUSSION THREAD

**Discussion by benhonda** [tailwindlabs/headlessui#1798] [Help]
The spacebar key-press gets interrupted by something in the `RadioGroup` component. The result is that I can't type a whitespace when my input is inside `RadioGroup`.

I was unable to diagnose what exactly is causing this issue but this got me wondering... is it bad practice to nest `<input>` inside `<RadioGroup>`?

As a hack-fix, I've added an `onKeyDown` event to my input:
```html
<RadioGroup ...>
...
<input ... onKeyDown={(e) => (e.key == " " || e.code == "Space" || e.keyCode == 32) && e.stopPropagation()} />
...
</RadioGroup>
```

**Comment by gcaraciolo** [comment]
For vuejs

```html
<RadioGroup ...>
...
<input ... @keydown.stop />
...
</RadioGroup>
```

**Comment by enzocomics** [comment]
I ran into this issue today with a `Textarea` that is revealed when a `Radio` inside a `RadioGroup` is selected. I'm also wondering if it's bad practice to do that. Pleased to report the hack-fix by @benhonda still works well, though. Thank you for that!
